www/js/menu.js

```diff
@@ -2,57 +2,68 @@ const h = require('hyperscript');
 const settings = require('./settings');
 const getJSON = require('get-json');
 
+const buttonFactory = (options) => {
+  let classList;
+  const btnId = options.buttonId || '';
+  const btnIcon = options.buttonIcon || 'fa-times';
+  if (options.buttonType === 'open') {
+    classList = `#${btnId}.active`;
+  } else {
+    classList = '.close-button';
+  }
+  return h(
+    `a${classList}.navigator-button`,
+    {
+      onclick: () => {
+        module.exports.toggleMenu(options.pageToToggle);
+      } },
+    h(`i.fa.${btnIcon}`));
+};
+
+const goToShabadPage = (shabadId) => {
+  global.core.search.loadShabad(shabadId);
+  module.exports.toggleMenu('#shabad-menu-page');
+  document.querySelector('#shabad-pageLink').click();
+};
+
+/* Generate Toggle Buttons */
 const menuButton = h(
   'a.menu-button.navigator-button.active',
   h('i.fa.fa-bars'));
-const closeButton = h(
-  'a.close-button.navigator-button',
-  {
-    onclick: () => {
-      module.exports.toggleMenu('#menu-page');
-    } },
-  h('i.fa.fa-times'));
-const customSlidesButton = h(
-  'a#custom-slide-menu.navigator-button.active',
-  {
-    onclick: () => {
-      module.exports.toggleMenu('#custom-slides-page');
-    } },
-  h('i.fa.fa-clone'));
-const shabadMenuButton = h(
-  'a#shabad-menu.navigator-button.active',
-  {
-    onclick: () => {
-      module.exports.toggleMenu('#shabad-menu-page');
-    } },
-  h('i.fa.fa-briefcase'));
-const shabadMenuCloseButton = h(
-  'a.close-button.navigator-button',
-  {
-    onclick: () => {
-      module.exports.toggleMenu('#shabad-menu-page');
-    } },
-  h('i.fa.fa-times'));
-const customSlidesCloseButton = h(
-  'a.close-button.navigator-button',
-  {
-    onclick: () => {
-      module.exports.toggleMenu('#custom-slides-page');
-    } },
-  h('i.fa.fa-times'));
+const customSlidesButton = buttonFactory({
+  buttonType: 'open',
+  buttonIcon: 'fa-clone',
+  buttonId: 'custom-slides-menu',
+  pageToToggle: '#custom-slides-page',
+});
+const shabadMenuButton = buttonFactory({
+  buttonType: 'open',
+  buttonIcon: 'fa-archive',
+  buttonId: 'shabad-menu',
+  pageToToggle: '#shabad-menu-page',
+});
+const closeButton = buttonFactory({
+  buttonType: 'close',
+  pageToToggle: '#menu-page',
+});
+const shabadMenuCloseButton = buttonFactory({
+  buttonType: 'close',
+  pageToToggle: '#shabad-menu-page',
+});
+const customSlidesCloseButton = buttonFactory({
+  buttonType: 'close',
+  pageToToggle: '#custom-slides-page',
+});
+
+/* load Shabad buttons */
 const randomShabadButton = h(
   'li',
   h(
     'a.random-shabad-button',
     {
       onclick: () => {
         global.platform.search.randomShabad()
-          .then((shabadId) => {
-            global.core.search.loadShabad(shabadId);
-            module.exports.toggleMenu('#shabad-menu-page');
-            // go to shabad page
-            document.querySelector('#shabad-pageLink').click();
-          });
+          .then((shabadId) => { goToShabadPage(shabadId); });
```

Comment:
No need to create extra function :D 

```diff
- .then((shabadId) => { goToShabadPage(shabadId); });
+ .then(goToShabadPage);
```

Reply:
Sorry, I didn't get your point. Basically right now a new function is created which takes the argument and gives it to goToShabadPage. For such scenario, we can simply pass function reference.

Reply:
Oh! Okay, got it! Thank you 🙏